### PR TITLE
[router] log user IPs

### PR DESCRIPTION
**Pull request by danking**
Gateway receives the user IP (thanks to #8045). However, gateway is an HTTP
proxy, so packets from gateway necessarily come from gateway's IP. Gateway
places the user IP into the HTTP header `X-Real-IP`. All downstream servers
must: log `X-Real-IP` and forward `X-Real-IP` unadulterated. This PR makes that
change for `router`.

- fix router Makefile (`domain` is now in `global`)
- add `proxy.conf` which configures the standard proxy headers (importantly:
  forwards `X-REAL-IP`)
- for non-notebook servers, `include` the `proxy.conf`
- for notebook, update to include proxy headers
- override default `access_log` (which required checking in the default
  `nginx.conf`)
- lift other `http` directives into `nginx.conf` now that it is checked in

### CODE REVIEW

**Comment by danking** [comment]
I copied `nginx.conf` from a running router and then trimmed out the comments and made our changes (see `access_log` and `log_format`).

File: router/router.nginx.conf.in [hunk 1]
```diff
@@ -205,22 +204,18 @@ server {
     listen [::]:80;
 }
 
-map $http_upgrade $connection_upgrade {
-    default upgrade;
-    ''      close;
```
Comment (jigold) [inline router/router.nginx.conf.in]: Why did you delete this?

File: router/router.nginx.conf.in [hunk 2]
```diff
@@ -1,15 +1,3 @@
-server_names_hash_bucket_size 128;
-
-map $http_x_forwarded_proto $updated_scheme {
-     default $http_x_forwarded_proto;
-     '' $scheme;
-}
```
Comment (jigold) [inline router/router.nginx.conf.in]: Why did you delete these? Is it because they're in nginx.conf?

Comment (danking) [reply router/router.nginx.conf.in]: Yes. I lifted all the `http` level directives into one place, including the new `access_log` and `log_format`.

**Comment by danking** [comment]
bump
